src/utils/c_utils/src/string_c_utils.c

```diff
@@ -422,50 +422,50 @@ bool MallocAndSubstr(char** target, char* source, size_t len)
  * @param src The source string to be copied.
  * @param destByteLen The size of the destination buffer in bytes, including plus 1 for the null terminator.
  * @param numSrcCharsToCopy The number of source chars to copy. It will be truncated and null-terminated if >= destByteLen.
- * @return int zero on success, or a negative number if there was an error.
+ * @return true on success.
  */
-int ADUC_Safe_StrCopyN(char* dest, const char* src, size_t destByteLen, size_t numSrcCharsToCopy)
+bool ADUC_Safe_StrCopyN(char* dest, const char* src, size_t destByteLen, size_t numSrcCharsToCopy)
 {
     size_t srcLen = 0;
 
     if (dest == NULL || src == NULL || destByteLen == 0)
     {
-        return -1;
+        return false;
     }
     *dest = '\0';
 
     if (numSrcCharsToCopy >= destByteLen)
     {
-        return -2;
+        return false;
     }
 
     srcLen = ADUC_StrNLen(src, numSrcCharsToCopy);
     if (numSrcCharsToCopy > srcLen)
     {
-        return -3;
+        return false;
     }
 
     memcpy(dest, src, numSrcCharsToCopy);
     dest[numSrcCharsToCopy] = '\0';
 
-    return 0;
+    return true;
 }
 
 /**
 * @brief Allocate a new str that is a copy of the src string with given max number of char.
 * @param dest The addr of ptr that will point to newly allocated string copy.
 * @param src The source string.
 * @param srcLen The char length of source string (not including null terminator)
-* @return 0 on success, -2 on invalid argument, and -1 for other errors.
+* @return true on success
 */
-int ADUC_AllocAndStrCopyN(char** dest, const char* src, size_t srcLen)
+bool ADUC_AllocAndStrCopyN(char** dest, const char* src, size_t srcLen)
 {
-    int res = -1;
+    bool res = false;
     char* tmpDest = NULL;
 
     if (dest == NULL || src == NULL || srcLen == 0)
```

Comment:
so this isn't allowed?

`AllocAndStrCopyN(&dest, "jeff", 0); // should return allocated `""`?`

Reply:
not currently as this function doesn't allocate on caller's behalf. For that, one can call other existing string_c_utils function above: MallocAndSubstr(&dest, "jeff", 0);